app/src/main/java/chat/rocket/android/fragment/server_config/InputHostnameFragment.java

```diff
@@ -37,11 +40,42 @@ protected void onSetupView() {
   }
 
   private void handleConnect() {
+    final String hostname = ServerPolicyHelper.enforceHostname(getHostname());
+
+    ServerPolicyHelper.isApiVersionValid(OkHttpHelper.getClientForUploadFile(), hostname,
+        new ServerPolicyHelper.Callback() {
+          @Override
+          public void isValid() {
+            getActivity().runOnUiThread(() -> onServerValid(hostname));
+          }
+
+          @Override
+          public void isNotValid() {
+            getActivity().runOnUiThread(() ->
+                Toast.makeText(getActivity(), R.string.input_hostname_invalid_server_message,
+                    Toast.LENGTH_SHORT).show());
+          }
+        });
+  }
+
+  @Override
+  public void onResume() {
```

Comment:
I didn't even see that. Was not in the part of the code I was changing.

We should probably do it the next time we touch that class. :)